### ISSUE TITLE
Crash on LSP diagnostic

### ISSUE THREAD

**Issue by dead10ck** [helix-editor/helix#2572]
### Summary

I experienced a crash of helix with this message:

```
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', /home/dead10ck/src/helix/helix-term/src/application.rs:459:81
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This was on d2b1add1f41c9d9a655a4637e3e276eb92e8eefb and the problematic issue is here:

https://github.com/helix-editor/helix/blob/d2b1add1f41c9d9a655a4637e3e276eb92e8eefb/helix-term/src/application.rs#L441-L459

It appears that it was trying to handle a diagnostic message before the LSP was fully initialized, and this code seems to assume that if we are handling this message, the LSP must be fully initialized.

### Helix Version

helix d2b1add1f41c9d9a655a4637e3e276eb92e8eefb

**Comment by dead10ck** [comment]
@frexsdev oh, I don't see any change? It still looks the same to me: https://github.com/helix-editor/helix/blob/82da9bd4f28cd2a050222173d6429478bed5b3cb/helix-term/src/application.rs#L465

Was it fixed somewhere else in the code?

**Comment by pascalkuthe** [comment]
This was fixed by  #7467
